flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -542,6 +543,29 @@ public void testUpdateUnknownInputChannel() throws Exception {
 		}
 	}
 
+	@Test
```

Comment:
In addition, do you think we should also cover the current case for closing `SingleInputGate` and `creatingInputGates` in different threads?

Reply:
I think we can rely on ConcurrentHashMap provided in constructor (I changed the approach a bit), then there is no need to test ConcurrentHashMap.